database/sql/src/main/scala/cromwell/database/slick/MetadataSlickDatabase.scala

```diff
@@ -35,7 +39,7 @@ class MetadataSlickDatabase(originalDatabaseConfig: Config)
   override def addMetadataEntries(metadataEntries: Iterable[MetadataEntry])
                                  (implicit ec: ExecutionContext): Future[Unit] = {
     val action = DBIO.seq(metadataEntries.grouped(insertBatchSize).map(dataAccess.metadataEntries ++= _).toSeq:_*)
-    runAction(action)
+    runTransaction(action)
```

Comment:
Why this change?

Reply:
Let me go back and double-check this.  It might be detritus from an earlier attempt that got much messier.

Reply:
Nope, this is what happens if I don't run transactionally:
```
2019-05-15 11:03:02,775 cromwell-system-akka.dispatchers.service-dispatcher-11 E
RROR - WriteMetadataActor Failed to properly process data
org.postgresql.util.PSQLException: Large Objects may not be used in auto-commit
mode.
        at org.postgresql.largeobject.LargeObjectManager.createLO(LargeObjectMan
ager.java:295)
        at org.postgresql.largeobject.LargeObjectManager.createLO(LargeObjectMan
ager.java:283)
        at org.postgresql.jdbc.PgPreparedStatement.setClob(PgPreparedStatement.j
ava:1188)
        at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.setClob(HikariPro
xyPreparedStatement.java)
        at slick.jdbc.JdbcTypesComponent$JdbcTypes$ClobJdbcType.setValue(JdbcTyp
esComponent.scala:149)
        at slick.jdbc.JdbcTypesComponent$JdbcTypes$ClobJdbcType.setValue(JdbcTyp
esComponent.scala:147)
        at slick.jdbc.OptionResultConverter.set(JdbcResultConverter.scala:33)
        at slick.jdbc.OptionResultConverter.set(JdbcResultConverter.scala:23)
        at slick.relational.ProductResultConverter.set(ResultConverter.scala:68)
        at slick.relational.ProductResultConverter.set(ResultConverter.scala:43)
        at slick.relational.TypeMappingResultConverter.set(ResultConverter.scala
:135)
        at slick.jdbc.JdbcActionComponent$InsertActionComposerImpl$MultiInsertAc
tion.$anonfun$run$15(JdbcActionComponent.scala:529)
        at slick.jdbc.JdbcActionComponent$InsertActionComposerImpl$MultiInsertAc
tion.$anonfun$run$15$adapted(JdbcActionComponent.scala:528)
        at scala.collection.Iterator.foreach(Iterator.scala:944)
        at scala.collection.Iterator.foreach$(Iterator.scala:944)
        at scala.collection.AbstractIterator.foreach(Iterator.scala:1432)
        at scala.collection.IterableLike.foreach(IterableLike.scala:71)
        at scala.collection.IterableLike.foreach$(IterableLike.scala:70)
        at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
        at slick.jdbc.JdbcActionComponent$InsertActionComposerImpl$MultiInsertAction.$anonfun$run$14(JdbcActionComponent.scala:528)
        at slick.jdbc.JdbcBackend$SessionDef.withPreparedStatement(JdbcBackend.scala:386)
        at slick.jdbc.JdbcBackend$SessionDef.withPreparedStatement$(JdbcBackend.scala:381)
        at slick.jdbc.JdbcBackend$BaseSession.withPreparedStatement(JdbcBackend.scala:448)
        at slick.jdbc.JdbcActionComponent$InsertActionComposerImpl.preparedInsert(JdbcActionComponent.scala:501)
        at slick.jdbc.JdbcActionComponent$InsertActionComposerImpl$MultiInsertAction.run(JdbcActionComponent.scala:526)
        at slick.jdbc.JdbcActionComponent$SimpleJdbcProfileAction.run(JdbcActionComponent.scala:30)
        at slick.jdbc.JdbcActionComponent$SimpleJdbcProfileAction.run(JdbcActionComponent.scala:27)
        at slick.dbio.DBIOAction$$anon$4.$anonfun$run$3(DBIOAction.scala:239)
        at scala.collection.Iterator.foreach(Iterator.scala:944)
        at scala.collection.Iterator.foreach$(Iterator.scala:944)
        at scala.collection.AbstractIterator.foreach(Iterator.scala:1432)
        at scala.collection.IterableLike.foreach(IterableLike.scala:71)
        at scala.collection.IterableLike.foreach$(IterableLike.scala:70)
        at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
        at slick.dbio.DBIOAction$$anon$4.run(DBIOAction.scala:239)
        at slick.dbio.DBIOAction$$anon$4.run(DBIOAction.scala:237)
        at slick.dbio.SynchronousDatabaseAction$$anon$7.run(DBIOAction.scala:486)
        at slick.basic.BasicBackend$DatabaseDef$$anon$2.liftedTree1$1(BasicBackend.scala:275)
        at slick.basic.BasicBackend$DatabaseDef$$anon$2.run(BasicBackend.scala:275)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
        at java.lang.Thread.run(Thread.java:748)
```
I'm convinced there is a bug in how `Clob` is handled by the Postgres driver, because it's just a `text` field in tables, which shouldn't require large object support.  Unfortunately I don't think there's an elegant workaround the way there was for the `SELECT` queries.

Reply:
Should I change it to run transactionally only for Postgres?

Reply:
Yes, please.

Transactions have semantic and performance effects. If the semantics don't require a transaction, I'd steer clear. We insert a ton of metadata and making the RDBMS do so in a transaction basically tells it, stage all of these things in a temporary area such that they can be committed with an atomic switch - instead of processing them as they come in. One can imagine that the latter is a lot easier.

Reply:
Okay, done.